patch/llpcConfigBuilderBase.cpp

```diff
@@ -346,6 +345,47 @@ void ConfigBuilderBase::SetPipelineHash()
     pipelineHashNode[1] = m_document->getNode(0U);
 }
 
+// =====================================================================================================================
+/// Append a single entry to the PAL register metadata.
+///
+/// @param [in] key The metadata key (usually a register address).
+/// @param [in] value The metadata value.
+void ConfigBuilderBase::AppendConfig(uint32_t key, uint32_t value)
+{
+    LLPC_ASSERT(key != InvalidMetadataKey);
+
+    Util::Abi::PalMetadataNoteEntry entry;
+    entry.key = key;
+    entry.value = value;
+    m_config.push_back(entry);
+}
+
+// =====================================================================================================================
+/// Append an array of entries to the PAL register metadata. Invalid keys are filtered out.
+///
+/// @param [in] config The array of register metadata entries.
+void ConfigBuilderBase::AppendConfig(llvm::ArrayRef<Util::Abi::PalMetadataNoteEntry> config)
+{
+    uint32_t count = 0;
+
+    for (const auto &entry : config)
```

Comment:
It's OK to place "&" and "*" just before the identifiers. Currently, I hope we can still follow old style "const auto&" until we find a timeslot to re-format existing codes.